core/store/src/config.rs

```diff
@@ -48,9 +48,13 @@ pub struct StoreConfig {
 }
 
 /// Mode in which to open the storage.
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Eq, PartialEq)]
 pub enum Mode {
+    /// Open an existing database in read-only mode.  Fail if it doesn’t exist.
     ReadOnly,
+    /// Open an existing database in read-write mode.  Fail if it doesn’t exist.
+    ReadWriteExisting,
+    /// Open a database in read-write mode.  create if it doesn’t exist.
     ReadWrite,
```

Comment:
`ReadWrite` and `ReadWriteCreate` feel a tiny bit better to me perhaps, but no strong opinion. If most of call-sites are RW, than it makes sense that it is a shorter default.

Reply:
I agree but yes I also think the ergonomics are in favour of the current naming.  ReadWriteExisting will be only used internally.